pkg/operator/apiservicecontroller/apiservice_controller.go

```diff
@@ -230,3 +255,93 @@ func (c *APIServiceController) eventHandler() cache.ResourceEventHandler {
 		DeleteFunc: func(obj interface{}) { c.queue.Add(workQueueKey) },
 	}
 }
+
+// TODO: change the name
+func (c *APIServiceController) filterAPIServices() []*apiregistrationv1.APIService {
+
+	oauthOperatorPrecondition := c.oauthOperatorAvailableAndVersionMatch()
+
+	filteredApiServices := []*apiregistrationv1.APIService{}
+
+	// the c.apiServices list is authoritative
+	// it can compete with oauth-server in case of errors but it could save us on downgrade (the current coao and an old cao)
+	for _, apiService := range c.apiServices {
+		if c.isAPIServicePotentiallyManagedByExternalServer(apiService) && oauthOperatorPrecondition && c.isAPIServiceAnnotatedByExternalServer(apiService) {
+			// TODO: log that this resource is being skipped on behalf of oauth-server
+			continue
+		}
+		filteredApiServices = append(filteredApiServices, apiService)
+	}
+
+	return filteredApiServices
+}
+
+func (c *APIServiceController) oauthOperatorAvailableAndVersionMatch() bool {
+	oauthOperator, err := c.clusterOperatorLister.Get("authentication")
+	if err != nil {
+		// TODO: log the err
+		return false
+	}
+	if oauthOperator == nil {
+		return false
+	}
+
+	if !clusteroperatorv1helpers.IsStatusConditionTrue(oauthOperator.Status.Conditions, configv1.OperatorAvailable) {
+		return false
+	}
+
+	operatorVersionString := c.versionRecorder.GetVersions()["operator"]
+	oauthOperatorVersionString := ""
+	for _, oauthOperatorVersion := range oauthOperator.Status.Versions {
+		if oauthOperatorVersion.Name == "operator" {
+			oauthOperatorVersionString = oauthOperatorVersion.Version
+			break
+		}
+	}
+	if len(oauthOperatorVersionString) == 0 || len(operatorVersionString) == 0 {
+		return false
+	}
+
+	oauthOperatorVersion, err := semver.Parse(oauthOperatorVersionString)
+	if err != nil {
+		// TODO: log the err
```

Comment:
nah, ripple the error all the way up.

Reply:
> nah, ripple the error all the way up.

Yeah, this won't work well even in CI builds.  those don't have real versions.

Reply:
won't returning an error put the operator into a degraded state?